### PR TITLE
Edited post proxy

### PR DESCRIPTION
**Pull request by planarvoid**
This PR replaces the use of `PostModel` with a synchronized `EditPostRepository` in the `EditPostActivity` and the related class. The goal here is to have a single source of access to modifying `PostModel` (which should be immutable). There should be no behavioral changes to the activity. The activity hook now exposes the repository instead of the `PostModel`

To test:
* Stress test post editing

PR submission checklist:

- [x] I have considered adding unit tests where possible.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by malinajirka** [comment]
Thanks @planarvoid ! I've reviewed the code and it LGTM. However, I've encountered a bug during my tests.

1. Open an existing published post
2. Add "abc" to the end of the post
3. Switch to classic/Gutenberg editor
4. Notice the "abc" you added in step 2 disappears

**Comment by planarvoid** [comment]
Thanks for the check @malinajirka ! I've found the bug that was causing the behaviour. I was doing `snapshot = post` instead of `snapshot = post.clone()` and because of this there were never changes found between the snapshot and the post 🤦‍♂

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -353,8 +355,9 @@ private void newPostSetup() {
         }
 
         // Create a new post
-        mEditPostRepository.setPost(mPostStore.instantiatePostModel(mSite, mIsPage, null, null));
-        mEditPostRepository.setStatus(PostStatus.DRAFT);
+        PostModel post = mPostStore.instantiatePostModel(mSite, mIsPage, null, null);
+        post.setStatus(PostStatus.DRAFT.toString());
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Yep, that's the reason. We can do what you're suggesting but this seemed like a simple enough solution.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostRepository.kt [hunk 2]
```diff
@@ -26,136 +27,60 @@ class EditPostRepository
...
-    private val snapshotStatus: PostStatus?
-        get() = lock.read {
-            postSnapshotWhenEditorOpened?.let { PostStatus.fromPost(it) }
-        }
 
     private val lock = ReentrantReadWriteLock()
 
-    fun clearLocation() = writeToPost { this.clearLocation() }
+    fun updateInTransaction(action: (PostModel) -> Boolean) = lock.write {
+        post?.let {
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostRepository.kt]: That makes sense 👍

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 3]
```diff
@@ -1560,47 +1576,48 @@ private void trackEditorCreatedPost(String action, Intent intent) {
...
-            postTitleOrContentChanged =
-                    updatePostContentNewEditor(isAutosave, (String) mEditorFragment.getTitle(),
-                            (String) mEditorFragment.getContent(mPost.getContent()));
-        }
-
-        // only makes sense to change the publish date and locally changed date if the Post was actually changed
-        if (postTitleOrContentChanged) {
-            PostUtils.updatePublishDateIfShouldBePublishedImmediately(mPost);
-            mPost.setDateLocallyChanged(DateTimeUtils.iso8601FromTimestamp(System.currentTimeMillis() / 1000));
+            return true;
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: As discussed, I've changed this to `return false`

**Comment by planarvoid** [comment]
@malinajirka thanks for the check!
I think I must've fixed the issue you pointed out in another change because it seems to work now: 

empty_post

**Pull request merged**